core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java

```diff
@@ -132,6 +203,26 @@ public HiveSqlDialect(Context context) {
     }
   }
 
+  @Override public SqlNode getCastCall(
+      SqlNode operandToCast, RelDataType castFrom, RelDataType castTo) {
+    if (castTo.getSqlTypeName() == SqlTypeName.TIMESTAMP && castTo.getPrecision() > 0) {
+      return new CastCallBuilder(this).makCastCallForTimestampWithPrecision(operandToCast,
+          castTo.getPrecision());
```

Comment:
Please create CastBuilder object once and use reference variable

Reply:
Done